src/main/java/com/hivemq/cli/commands/hivemq/schemas/CreateSchemaCommand.java

```diff
@@ -0,0 +1,145 @@
+
+    @SuppressWarnings({"NotNullFieldNotInitialized", "unused"})
+    @CommandLine.Option(names = {"--type"},
+                        required = true,
+                        converter = SchemaTypeConverter.class,
+                        description = "the schema type (default json) ")
+    private @NotNull String schemaType;
+
+    @SuppressWarnings("unused")
+    @CommandLine.Option(names = {"--message-type"},
+                        description = "the Protobuf message type. Only used with --type protobuf.")
+    private @Nullable String messageType;
+
+    @SuppressWarnings("unused")
+    @CommandLine.Option(names = {"--allow-unknown"},
+                        defaultValue = "false",
+                        description = "allow unknown Protobuf fields (default false). Only used with --type protobuf.")
+    private boolean allowUnknown;
+
+    @SuppressWarnings({"NotNullFieldNotInitialized", "unused"})
+    @CommandLine.ArgGroup(multiplicity = "1")
+    private @NotNull SchemaDefinitionOptions definitionOptions;
+
+    @CommandLine.Mixin
+    private final @NotNull DataGovernanceOptions dataGovernanceOptions = new DataGovernanceOptions();
+
+    private final @NotNull OutputFormatter outputFormatter;
+    private final @NotNull HiveMQRestService hiveMQRestService;
+
+    @Inject
+    public CreateSchemaCommand(
+            final @NotNull HiveMQRestService hiveMQRestService,
+            final @NotNull OutputFormatter outputFormatter) {
+        this.outputFormatter = outputFormatter;
+        this.hiveMQRestService = hiveMQRestService;
+    }
+
+    @Override
+    public @NotNull Integer call() throws IOException {
+        Logger.trace("Command {}", this);
+
+        final SchemasApi schemasApi =
+                hiveMQRestService.getSchemasApi(dataGovernanceOptions.getUrl(), dataGovernanceOptions.getRateLimit());
+
+        if (schemaType.equals("protobuf") && messageType == null) {
+            outputFormatter.printError("Protobuf message type is missing. Option '--message-type' is not set.");
+            return 1;
+        }
+
+        if (schemaType.equals("json") && messageType != null) {
+            outputFormatter.printError("Option '--message-type' is not applicable to schemas of type 'json'.");
+            return 1;
+        }
+
+        final String fileDefinition = definitionOptions.getFile();
+        final ByteBuffer argumentDefinition = definitionOptions.getArgument();
+        final ByteBuffer definitionBytes;
+        if (fileDefinition != null) {
+            try {
+                final Path path = Paths.get(fileDefinition);
+                definitionBytes = ByteBuffer.wrap(Files.readAllBytes(path));
+            } catch (final IOException exception) {
+                outputFormatter.printError("File not found or not readable: " + fileDefinition);
+                return 1;
+            }
+        } else if (argumentDefinition != null) {
+            definitionBytes = argumentDefinition;
+        } else {
```

Comment:
Here when both `--file` and `--definition` are set:
- no error is thrown, but `--definition` wins
- but if a file is not found/readable -> exception is thrown

Maybe it is better to just thrown an exception if both are set saint that "One and only one of --file or --definition must be set."

Reply:
This is already handled by picocli because of the `@CommandLine.ArgGroup(multiplicity = "1")` annotation on `definitionOptions`

You will get
```
Error: --file=<file>, --definition=<argument> are mutually exclusive (specify only one)
Try 'mqtt hivemq schemas create --help' for more information.
```